planner/core/rule_join_reorder.go

```diff
@@ -507,6 +507,13 @@ func (s *baseSingleGroupJoinOrderSolver) makeJoin(leftPlan, rightPlan LogicalPla
 	remainOtherConds, otherConds = expression.FilterOutInPlace(remainOtherConds, func(expr expression.Expression) bool {
 		return expression.ExprFromSchema(expr, mergedSchema)
 	})
+	if (joinType.JoinType == LeftOuterJoin || joinType.JoinType == RightOuterJoin) && len(otherConds) > 0 {
```

Comment:
How about left outer semi join and anti left outer ？

Reply:
good